plugins/hls-tactics-plugin/test/golden/AutoThetaGADTDestruct.hs.expected

```diff
@@ -0,0 +1,7 @@
+{-# LANGUAGE GADTs #-}
+
+data X f = Monad f => X
+
+fun1 :: X f -> a -> f a
+fun1 X a = pure a
```

Comment:
I don't understand why this doesn't get eta-reduced

Reply:
FWIW on master this doesn't eta reduce either:
```Haskell
test :: Monad m => () -> a -> m a
test = _
```
But this does:
```Haskell
test2 :: Monad m => b -> a -> m a
test2 = _
```
So it may be caused by matching on a constructor (which X ofc is)

Reply:
A mystery for another day. Thanks for the simpler repro!